pkg/file/temp_dir_generator.go

```diff
@@ -1,50 +1,65 @@
 package file
 
 import (
-	"fmt"
-	"io/ioutil"
 	"os"
-	"sync"
+	"strings"
 
 	"github.com/hashicorp/go-multierror"
 )
 
 type TempDirGenerator struct {
-	tempDir []string
-	lock    *sync.Mutex
+	rootPrefix   string
+	rootLocation string
+	generators   []*TempDirGenerator
```

Comment:
nit: Consider renaming to something like `childGenerators` to make it more clear what the relationship is with the parent struct and why this field exists

Reply:
renamed to `children` 👍